client-v2/src/redux/modules/pageViewData/selectors.ts

```diff
@@ -1,34 +1,43 @@
 import { State } from '../../../types/State';
 import {
   PageViewDataPerFront,
-  PageViewDataPerCollection
+  PageViewDataPerCollection,
+  PageViewStory,
+  PageViewArticlesOnFront
 } from 'shared/types/PageViewData';
 
-const selectPageViewDataForArticlePath = (state: State, url: string) =>
-  selectPageViewData(state);
-
-const selectPageViewData = (state: State): PageViewDataPerFront[] =>
-  state.shared.pageViewData;
-
-const selectPageViewDataForCollection = (
+const selectPageViewDataForArticleId = (
   state: State,
-  collectionId: string,
+  articleId: string,
   frontId: string
-): PageViewDataPerCollection | undefined => {
-  if (!state.shared.pageViewData) {
-    return;
-  }
-  const possibleFront = state.shared.pageViewData.find(
-    front => front.frontId === frontId
-  );
+): PageViewStory | undefined => {
+  const dataForFront: PageViewDataPerFront | undefined = selectPageViewData(
+    state
+  ).find(front => front.frontId === frontId);
+
+  const allArticlesOnFront =
+    dataForFront &&
+    dataForFront.collections.reduce(
+      (acc: PageViewArticlesOnFront, curr: PageViewDataPerCollection) => {
```

Comment:
We'll need to memoize this to ensure we're not doing expensive graph work unnecessarily -- happy to pair!